read.go

```diff
@@ -125,29 +125,26 @@ func (r *reader) readTOC(toc *indexTOC) error {
 			if err != nil {
 				return err
 			}
+
 			sec := secs[tag]
-			if sec != nil && sec.kind() == sectionKind(kind) {
-				// happy path
-				if err := sec.read(r); err != nil {
-					return err
+			if sec == nil || sec.kind() != sectionKind(kind) {
+				// If we don't recognize the section, we may be reading a newer index than the current version. Use
+				// a "dummy section" struct to skip over it.
+				log.Printf("encountered malformed index section (%s), skipping over it", tag)
```

Comment:
is malformed the correct word to use here given it may just be us reading a backwards compatible shard?

Reply:
Good point, I'll say something like 'unrecognized' instead.